### ISSUE TITLE
autocomplete for keywords in YAML frontmatter

### ISSUE THREAD

**Issue by h-larsson** [Zettlr/Zettlr#4070]
### Checklist

- [X] Zettlr does not possess this feature (if it does, but the feature doesn't work out for you, propose an enhancement instead.)
- [X] Zettlr's current features cannot be used to approximate this behaviour (otherwise, propose an enhancement instead.)
- [X] I have searched the documentation to make sure I did not overlook it.
- [X] I have searched both open and closed issues to see if someone has already proposed this.

### Feature Proposal

Autocomplete for keywords in YAML frontmatter would be great. It already works for tags and 

### Example Workflow

Write 
```
---
title: "Welcome to Zettlr!"
keywords:
  - general
...
#genius 
```

in `file1.md`.
Start writing 
```
---
title: "Welcome to Zettlr! 2"
keywords:
  - gen
 ...
```
in `file2.md` and getting autocomplete suggestion for `gen` to either `general` or `genius`. 
(As far as I understand, tags and keywords are treated the same way. If that is not the case, the autocomplete should only suggest `general`). 

### Caveats

I cannot think of any. 

### Do you wish to attempt implementing this yourself?

No

### Platforms

- [X] Windows
- [X] macOS
- [X] Linux

### Additional Information

(I only tested this on linux with version 2.3.0 UUID: f77f0308-2ef0-4327-875a-0679c5d8c261)

**Comment by h-larsson** [comment]
According to boring-cyborg, this is not a feature request but an enhancement. How can I change the label?

**Comment by exanrauzer** [comment]
This is tricky case. The main difference between tags and keywords is that keyword can 'contain' two or more words (long-tail keyword), and tag can't (#or-it-can-like-that).

I think it could be considered if a keyword in yaml for example: `knowledge management software` could be taken as the same as #knowledge-management-software. Or maybe allow: #[knowledge management software] syntax, or other way.

But i feel that keyword is not the same 'thing' as tag. Tag is rather a: _link_ and _label_. And keyword is rather a... keyword, or even category(!) or container(!). You can have technical (functional) tags like: #draft #new #fix #dilemma which doesn't say anything about the context so I wouldn't put them in the same basket with keywords.

**Comment by h-larsson** [comment]
Thanks for the explanation. This may be a separate issue, but then I find it confusing that keywords *do* seem to be converted to tags -- they appear in the tag cloud as well as in the file manager. So to some extend the #[knowledge management software] syntax is already implicitly used.

![keyword](https://user-images.githubusercontent.com/10084569/213531196-db5bb5b1-bfc3-4770-9833-900bd07fe30a.png)

**Comment by exanrauzer** [comment]
Yes, well spotted. So they _are_ converted/combined with tags, however it is the end of the road. You cannot use the multi word tag created by keyword in text (body text). They are suggested but you cannot use them - they are broken. For example: `#knowledge management software` tag created by the keyword splits into three words with a <#> before the first one. Anyway this is not the point of your issue, sorry for digressions!

You suggests that autocompletion for keywords in yaml would be great, but what you literally suggest is to have autocompletion works after hypen, space and first typed letter, right?

**Comment by h-larsson** [comment]
Yes, thanks for the clarification. Sorry, I realized there is a typo in the issue. I'll change it.
